tools/pipelines/templates/include-publish-npm-package-steps.yml

```diff
@@ -138,6 +138,51 @@ steps:
                   exit 1
                 fi
             else
+              feedId=$(grep -oP '_packaging/\K[^/]+(?=/npm/)' publish_log)
+
+              if [[ $release == "release" && $feedId == "build"]]; then
+                packageVersion=$(grep -oP 'npm notice version:\s+\K[^\s]+' publish_log)
+                
+                if [[ "$packageName" != @(tinylicious|fluid-framework) ]]; then
+                  prefixes=("fluidframework" "fluid-tools" "fluid-internal" "fluid-experimental")
+                  for prefix in "${prefixes[@]}"; do
+                      if [[ "$packageName" == "$prefix"* ]]; then
+                          packageName="@${packageName/$prefix-/$prefix/}"
+                      fi
+                  done
+                fi
+
+                buildFeedBaseUrl="https://pkgs.dev.azure.com/fluidframework/internal/_apis/packaging/feeds/build/npm"
+                buildFeedPromotionUrl="$buildFeedBaseUrl/$packageName/versions/$packageVersion?api-version=7.1-preview.1"
```

Comment:
Just double checking if it's okay to expose this URL publicly

Reply:
Those should be variables instead of hard coded.

Reply:
Also, you might need to deal with the "dev" feed as well?

Reply:
@curtisman It looks like the "internal" feed(the feed that 1P customers pull packages from) aggregates only from "build" and "test" feeds. Since branches with test/* will not be important to 1P customers we should only promote packages from the build feed.

relevant links showing the upstream sources;
https://eng.ms/docs/experiences-devices/opg/office-shared/fluid-framework/fluid-framework-internal/fluid-framework/docs/infrastructure/ado/feeds
https://dev.azure.com/fluidframework/internal/_artifacts/feed/dev/settings/upstreams.

Reply:
> Just double checking if it's okay to expose this URL publicly

we have historically and never purged git history so it's not really a new attack vector. A malicious actor would need to have collaborator access to these feeds to do anything--e.g. via a leaked PAT. We've been moving toward controlling these things with variables more for code cleanliness than security reasons.

Reply:
> Also, you might need to deal with the "dev" feed as well?

@RishhiB did you intend to resolve this bit? It does seem relevant. You have access to this in the template parameters as `devFeedName`.

Reply:
@Abe27342 did I understand @curtisman incorrectly? I believe dev does not need to be handled because we are only promoting the release package on the build feed. More explanation in this comment; https://github.com/microsoft/FluidFramework/pull/16610#discussion_r1304616447

Reply:
right, but we're hooking into the "publish packages for scoped" step, which publishes packages to multiple registries (@fluid-internal packages get published to dev as part of this step). it's not obvious to me how this is handled--might just be missing something. If we're not trying to invoke the ADO REST API at all for fluid-internal packages, I think it's fine for now.